src/Lookup.hs

```diff
@@ -28,6 +29,27 @@ lookupInEnv path@(SymPath (p : ps) name) env =
         Just parent -> lookupInEnv path parent
         Nothing -> Nothing
 
+lookupInterfaceOrType :: Context -> SymPath -> Maybe Binder
+lookupInterfaceOrType ctx path =
+  let typeEnv = (getTypeEnv (contextTypeEnv ctx))
+   in lookupBinder path typeEnv
+
+lookupGlobalBinding :: Context -> SymPath -> Maybe Binder
```

Comment:
Wouldn't it make sense to prefix these with something that makes them a bit different from the "old" lookup functions (or add something to the old ones, like a "InEnv" suffix or similarly) ? Right now it will be very hard to anticipate what type each lookup function expects when reading the code.

Reply:
4d30c24